src/google-maps/google-map/google-map.ts

```diff
@@ -17,28 +34,93 @@ import {ReplaySubject} from 'rxjs';
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: '<div class="map-container"></div>',
 })
-export class GoogleMap implements OnInit {
-  // Arbitrarily chosen default size
-  @Input() height = '500px';
-  @Input() width = '500px';
+export class GoogleMap implements OnInit, OnDestroy {
+  @Input()
+  set height(height: string) {
+    this._height$.next(height || DEFAULT_HEIGHT);
+  }
 
-  // TODO(mbehrlich): add options, handlers, properties, and methods.
+  @Input()
+  set width(width: string) {
+    this._width$.next(width || DEFAULT_WIDTH);
+  }
 
-  private readonly _map$ = new ReplaySubject<google.maps.Map>(1);
+  @Input()
+  set center(center: google.maps.LatLngLiteral) {
+    this._center$.next(center);
+  }
+  @Input()
+  set zoom(zoom: number) {
+    this._zoom$.next(zoom);
+  }
+  @Input()
+  set options(options: google.maps.MapOptions) {
+    this._options$.next(options || DEFAULT_OPTIONS);
+  }
 
-  constructor(private readonly _elementRef: ElementRef) {}
+  // TODO(mbehrlich): Add event handlers, properties, and methods.
 
-  ngOnInit() {
-    // default options set to the Googleplex
-    const options: google.maps.MapOptions = {
-      center: {lat: 37.421995, lng: -122.084092},
-      zoom: 17,
-    };
+  private readonly _height$ = new BehaviorSubject<string>(DEFAULT_HEIGHT);
+  private readonly _width$ = new BehaviorSubject<string>(DEFAULT_WIDTH);
+  private readonly _options$ = new BehaviorSubject<google.maps.MapOptions>(DEFAULT_OPTIONS);
+  private readonly _center$ = new BehaviorSubject<google.maps.LatLngLiteral|undefined>(undefined);
+  private readonly _zoom$ = new BehaviorSubject<number|undefined>(undefined);
```

Comment:
This whole setup with the subjects and the calls to `combineLatest`seems a bit over-complicated to me. What would be the problem with doing the initial set up in `ngOnInit` with the current values, and then updating the options/width/height in `ngOnChanges`?

Reply:
In general I prefer the setter pattern to ngOnChanges because I don't like the lack of types in the change interface and I don't like to make a complicated series of if blocks to differentiate various changes. I think this will especially become an issue for this component as we start attaching marker and infoWindow components to the map (among other features). If this isn't readable enough I would prefer to separate out helper methods and/or comment better. I've added comments for now.

Reply:
What does the maps API do if you call something like `setOptions` / `setSize` without the values changing? I'm wondering if just always calling the methods with the current state would be an option (without needing to check the changes structure at all).

Reply:
I don't think with setting the size, but there is a problem with setOptions, which is that if the user pans the googleMap and then changeDetection occurs for a reason other than the options changing, the map will reset the options back to whatever was last set explicitly.

Reply:
Ah, that makes sense, then. In that case, what do you think about just using `ngOnChanges` for the size parts? For the rest, I think it would be good to break the majority of the logic out of the lifecycle hook directly and into smaller functions; in general I prefer putting hook behavior in other functions just so that eachbehavior has a name ([kind of like this](https://github.com/angular/components/blob/615070acbdb2962eca3bebadb19720c614848981/src/cdk-experimental/popover-edit/table-directives.ts#L211)).

Reply:
Sounds good. Done.